modules/javafx.base/src/test/java/test/javafx/collections/ObservableMapTest.java

```diff
@@ -182,6 +180,21 @@ public void testPutAll(Callable<ObservableMap<String, String>> mapFactory) throw
         observer.assertMultipleCalls(call("oFoo", null, "OFoo"), call("pFoo", null, "PFoo"), call("foo", "bar", "foofoo"));
     }
 
+    @ParameterizedTest
+    @MethodSource("createParameters")
+    @SuppressWarnings("unchecked")
+    public void testReplaceAll(Callable<ObservableMap<String, String>> mapFactory) throws Exception {
+        setUp(mapFactory);
+
+        observableMap.replaceAll((key, value) -> switch (key) {
+            case "one" -> "2";
+            case "two" -> "3";
+            default -> value;
+        });
+
+        observer.assertMultipleCalls(call("one", "1", "2"), call("two", "2", "3"));
+    }
```

Comment:
From the description:

```
The implementation is fully backwards-compatible for listeners that
are unaware of the new API. If the next() method is not called, then
all subsequent changes are delivered as usual by repeated listener
invocations.

If a listener only fetches some changes of a bulk operation (but stops
halfway through the operation), the remaining changes will also be
delivered with repeated listener invocations.
```

1. should this be included in javadoc?
2. could we add a test for this scenario?